tools/jsdoc_checker.js

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env -S deno run --allow-read --allow-env --allow-sys
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+import { Node, Project, ts } from "npm:ts-morph@22.0.0";
+import { join, ROOT_PATH } from "./util.js";
+
+const libs = [
+  join(ROOT_PATH, "ext/cache/lib.deno_cache.d.ts"),
+  join(ROOT_PATH, "ext/console/lib.deno_console.d.ts"),
+  join(ROOT_PATH, "ext/url/lib.deno_url.d.ts"),
+  join(ROOT_PATH, "ext/web/lib.deno_web.d.ts"),
+  join(ROOT_PATH, "ext/fetch/lib.deno_fetch.d.ts"),
+  join(ROOT_PATH, "ext/websocket/lib.deno_websocket.d.ts"),
+  join(ROOT_PATH, "ext/webstorage/lib.deno_webstorage.d.ts"),
+  join(ROOT_PATH, "ext/canvas/lib.deno_canvas.d.ts"),
+  join(ROOT_PATH, "ext/crypto/lib.deno_crypto.d.ts"),
+  join(ROOT_PATH, "ext/net/lib.deno_net.d.ts"),
+  join(ROOT_PATH, "cli/tsc/dts/lib.deno.ns.d.ts"),
+  join(ROOT_PATH, "cli/tsc/dts/lib.deno.shared_globals.d.ts"),
+  join(ROOT_PATH, "cli/tsc/dts/lib.deno.window.d.ts"),
+];
+
+const unstableLibs = [
+  join(ROOT_PATH, "ext/broadcast_channel/lib.deno_broadcast_channel.d.ts"),
+  join(ROOT_PATH, "cli/tsc/dts/lib.deno_webgpu.d.ts"),
+  join(ROOT_PATH, "cli/tsc/dts/lib.deno.unstable.d.ts"),
+];
+
+const errors = [];
+
+const project = new Project();
+project.addSourceFilesAtPaths(libs);
+const unstableFiles = project.addSourceFilesAtPaths(unstableLibs);
+
+for (const file of project.getSourceFiles()) {
+  for (
+    const node of file.getDescendants().filter((descendant) =>
+      Node.isExportable(descendant)
+    )
+  ) {
+    if (
+      node.getKind() === ts.SyntaxKind.ModuleDeclaration &&
+      node.getName() === "Deno"
+    ) {
+      continue;
+    }
+
+    /*if (!(node.isExported() || node.hasDeclareKeyword())) {
+      throw new Error(getErrorPrefix(node) + "declare or export keyword");
+    }*/
```

Comment:
not sure if its necessary to enforce these, but its wildly inconsistent throughout the codebase

Reply:
It's probably best to standardize them.